### ISSUE TITLE
Validation and Evaluation not cumputed in run_qa.py

### ISSUE THREAD

**Issue by mamoon115** [huggingface/transformers#12978]
## Environment info

- `transformers` version: 4.10.0.dev0
- Platform: ubuntu
- Python version:3.8

Models:

- roberta

Library:
- trainer: @sgugger
- pipelines: @LysandreJik

## Information

Model I am using  is Roberta.

* while running run_qa.py on squad 2.0 data there is missing training metrics, validation, and evaluation loss. 
* There is training loss and evaluation metric display after finetuning.

The tasks I am working on is:
* an official QuestionAnswer/Squad task

**Comment by sgugger** [comment]
What command did you use to run the script? Setting `--eval_strategy epoch` for instance will give you the evaluation every epoch.

**Comment by mamoon115** [comment]
Yes
